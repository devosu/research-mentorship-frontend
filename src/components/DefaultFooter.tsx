// @components/DefaultFooter
//
// DefaultFooter component definition.

// NextJS essential imports.

// Local imports.

export default function DefaultFooter () {
	return (
		<footer className="bg-zinc-50 text-center dark:bg-neutral-700 lg:text-left" style={{bottom: 0, left: 0, right: 0, position: 'fixed'}}>
			<div className="bg-black/5 p-4 text-center text-surface dark:text-white">
				Â© 2024 Copyright:
				<a href="https://usg.osu.edu/"> Undergraduate Student Government at The Ohio State University</a>
			</div>
		</footer>
	);
}