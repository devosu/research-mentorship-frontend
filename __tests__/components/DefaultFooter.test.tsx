// Unit tests for DefaultFooter component.

// Testing essential imports.
import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";

// Local imports.
import DefaultFooter from "@src/components/DefaultFooter";

// Behavior for the DefaultFooter component.
describe("The default footer component", () => {
  it("renders a footer element", () => {
    const { container } = render(<DefaultFooter />);
    expect(container.querySelector("footer")).toBeDefined();
  });

  it("matches the snapshot", () => {
    const { container } = render(<DefaultFooter />);
    expect(container).toMatchSnapshot();
  });
});
