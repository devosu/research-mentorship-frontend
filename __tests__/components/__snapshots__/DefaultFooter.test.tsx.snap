// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The default footer component matches the snapshot 1`] = `
<div>
  <footer
    class="bg-zinc-50 text-center dark:bg-neutral-700 lg:text-left"
  >
    <div
      class="bg-black/5 p-4 text-center text-surface dark:text-white"
    >
      Â© 2024 Copyright:
      <a
        href="https://usg.osu.edu/"
      >
         Undergraduate Student Government at The Ohio State University
      </a>
    </div>
  </footer>
</div>
`;
